### PR TITLE
MTL-2247 partial -- adds gitea role

### PR DESCRIPTION
**Pull request by jpdavis-prof**
### Summary and Scope

- adds ansible role for installing and configuring **gitea** (written by @manderson-hpe, stolen by @jpdavis-prof)

#### Issue Type

- RFE Pull Request

### Prerequisites

- [x] I have included documentation in my PR (or it is not required)
- [x] I have tested this by building an image using this branch HEAD, or this is does not pertain to the node-images pipeline.
- [x] I tested this on internal system or this pertains to the node-images pipeline (if yes and not build related, please include results or a description of the test)
 
### Idempotency
 
Yes
 
### Risks and Mitigations
 
None

### CODE REVIEW

File: roles/gitea/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+gitea_dl_url: "https://github.com/go-gitea/gitea/releases/download/v{{ gitea_version }}/gitea-{{ gitea_version }}-linux-{{ gitea_arch }}"
+gitea_repository_root: "{{ gitea_home }}"
+gitea_user_repo_limit: -1
+
+gitea_app_name: "Metal Gitea"
+gitea_http_domain: localhost
+gitea_root_url: http://localhost:3000
+gitea_protocol: http
+gitea_http_listen: 0.0.0.0
+gitea_http_port: 3000
```
Comment (rustydb) [inline roles/gitea/defaults/main.yml:19]: Use `"{{ port_gitea }}"

File: roles/gitea/defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+gitea_disable_http_git: false
+gitea_offline_mode: true
+
+gitea_db_type: sqlite3
+gitea_db_path: "{{ gitea_home }}/data/gitea.db"
+
+gitea_ssh_listen: 0.0.0.0
+gitea_ssh_domain: gitea
+gitea_start_ssh: true
+gitea_ssh_port: 2222
```
Comment (rustydb) [inline roles/gitea/defaults/main.yml:29]: What is this used for, cloning with SSH?
If it's used for cloning/pulling, maybe add a new var into `vars/fawkes.yml` for this.

File: roles/gitea/defaults/main.yml [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+gitea_secret_key: T0pS3cr31
+gitea_internal_token: SomethingVeryLong
```
Comment (rustydb) [inline roles/gitea/defaults/main.yml:32]: Do we need to variablize these?

In the future, should these values be ansible-vault secrets or something?

File: roles/gitea/defaults/main.yml [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+gitea_repo_indexer_exclude: ""
+gitea_repo_exclude_vendored: true
+gitea_repo_indexer_max_file_size: 1048576
+
+gitea_log_level: Info
+
+gitea_extra_config: ""
+
+gitea_backup_on_upgrade: false
+gitea_backup_location: "{{ gitea_home }}/backups/"
```
Comment (rustydb) [inline roles/gitea/defaults/main.yml:79]: We should make a JIRA for mounting a backup volume or something here, so we can more easily reattach the gitea stuff to a new management-vm.

File: roles/gitea/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+  ansible.builtin.group:
+    name: "{{ gitea_group }}"
+    system: true
+    state: "present"
+
+- name: Create gitea user
+  ansible.builtin.user:
+    name: "{{ gitea_user }}"
+    home: "{{ gitea_home }}"
+    system: true
```
Comment (rustydb) [inline roles/gitea/tasks/main.yml:32]: Does `system: true` imply it won't have a home directory created? Do we need to set `create_home: false`?

or does it need a home directory?

Comment (rustydb) [reply roles/gitea/tasks/main.yml:32]: Nevermind, looking below I seemed to answer my own question.

File: roles/gitea/defaults/main.yml (hunk 1 above)
Comment (jpdavis-prof) [reply roles/gitea/defaults/main.yml:19]: vars_files exists.
```
- name: Node Images Management VM
  hosts: management_vm
  become: true # Enable builds that do not login as root to escalate privilege.
  vars_files:
    - vars/fawkes.yml
```

File: roles/gitea/defaults/main.yml (hunk 4 above)
Comment (jpdavis-prof) [reply roles/gitea/defaults/main.yml:79]: https://jira-pro.it.hpe.com:8443/browse/MTL-2263

**Review by rustydb — APPROVED** [review]

**Pull request merged**